### PR TITLE
[SPARK-50485][SQL] Unwrap SparkThrowable in (Unchecked)ExecutionException thrown by tableRelationCache

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR unwraps SparkThrowable in (Unchecked)ExecutionException thrown by tableRelationCache
### Why are the changes needed?

The guava cache library wraps exceptions thrown by `c: Callable[LogicalPlan]` as `(Unchecked)ExecutionException`s. This makes our code paths of special handling for SparkThrowable inoperative. For example, this kind of error in spark-sql cli is very lengthy.

### Does this PR introduce _any_ user-facing change?

Yes, when you visit an invalid plan in the table cache layer, you will get the original spark error instead of getting them under `(Unchecked)ExecutionException`s

### How was this patch tested?
new tests

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetLocationSuite.scala [hunk 1]
```diff
@@ -134,6 +135,23 @@ trait AlterTableSetLocationSuiteBase extends command.AlterTableSetLocationSuiteB
           "tableName" -> "`spark_catalog`.`ns`.`tbl`"))
     }
   }
+
+  test("SPARK-50485: Unwrap SparkThrowable in UEE thrown by tableRelationCache") {
+    withNamespaceAndTable("ns", "tbl") { t =>
+      sql(buildCreateTableSQL(t))
+      sql(s"CREATE TABLE t (a INT)")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetLocationSuite.scala:142]:
```suggestion
      sql(s"CREATE TABLE $t (a INT)")
```

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetLocationSuite.scala [hunk 2]
```diff
@@ -134,6 +135,23 @@ trait AlterTableSetLocationSuiteBase extends command.AlterTableSetLocationSuiteB
...
+      checkError(
+        exception = intercept[SparkUnsupportedOperationException] {
+          sql(s"ALTER TABLE t SET LOCATION 'https://mister/spark'")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetLocationSuite.scala:145]:
```suggestion
          sql(s"ALTER TABLE $t SET LOCATION 'https://mister/spark'")
```

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 3]
```diff
@@ -1258,6 +1258,22 @@ class QueryExecutionErrorsSuite
       )
     )
   }
+
+  test("SPARK-50485: Unwrap SparkThrowable in UEE thrown by tableRelationCache") {
+    withTable("t") {
+      sql(s"CREATE TABLE t (a INT)")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:1264]: s is not needed:
```suggestion
      sql("CREATE TABLE t (a INT)")
```

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 4]
```diff
@@ -1258,6 +1258,22 @@ class QueryExecutionErrorsSuite
...
+      checkError(
+        exception = intercept[SparkUnsupportedOperationException] {
+          sql(s"ALTER TABLE t SET LOCATION 'https://mister/spark'")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:1267]:
```suggestion
          sql("ALTER TABLE t SET LOCATION 'https://mister/spark'")
```

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 5]
```diff
@@ -1258,6 +1258,22 @@ class QueryExecutionErrorsSuite
...
+        },
+        condition = "FAILED_READ_FILE.UNSUPPORTED_FILE_SYSTEM",
+        parameters = Map(
+          "path" -> "https://mister/spark",
+          "fileSystemClass" -> "org.apache.hadoop.fs.http.HttpsFileSystem",
+          "method" -> "listStatus"))
+      sql(s"ALTER TABLE t SET LOCATION '/mister/spark'")
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:1274]:
```suggestion
      sql("ALTER TABLE t SET LOCATION '/mister/spark'")
```

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @yaooqinn.

**Comment by yaooqinn** [comment]
Thank you, @MaxGekk

**Pull request closed without merging**